File: business/business-oj/src/main/java/com/glowxq/oj/meta/enums/TextBusinessType.java
package com.glowxq.oj.meta.enums;

import com.glowxq.core.common.enums.base.BaseType;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum TextBusinessType implements BaseType {

    Announcement("Announcement", "公告"),
    Temp("Temp", "临时"),

    ;

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static TextBusinessType matchCode(String code) {
        for (TextBusinessType pushStatus : TextBusinessType.values()) {
            if (pushStatus.getCode().equals(code)) {
                return pushStatus;
            }
        }
        return null;
    }
}

File: oj-web/src/enums/oj/meta/TextBusinessType.ts
import type { BaseEnum } from '@/enums/base';

// 文本业务类型枚举接口
interface TextBusinessTypeEnum extends BaseEnum {
}

/**
 * 文本业务类型枚举
 * 用于标识不同的文本业务类型
 */
export const TextBusinessType: { [key: string]: TextBusinessTypeEnum } = {
  /**
   * 公告
   */
  ANNOUNCEMENT: {
    code: 'Announcement',
    name: '公告'
  },

  /**
   * 临时
   */
  TEMP: {
    code: 'Temp',
    name: '临时'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: string): TextBusinessTypeEnum | null {
  return Object.values(TextBusinessType).find(item => item.code === code) || null;
} 